gr-utils/python/modtool/modtool_newmod.py

```diff
@@ -102,3 +98,23 @@ def run(self, options):
             print("Created repository... you might want to commit before continuing.")
         print("Use 'gr_modtool add' to add a new block to this currently empty module.")
 
+
+### COMMAND LINE INTERFACE ###
+@click.command('newmod', short_help=ModToolNewModule().description)
+@click.option('--srcdir',
+              help="Source directory for the module template.")
+@ModTool.common_params
+@click.argument('module_name', metavar="MODULE-NAME", nargs=1, required=False)
+def cli(**kwargs):
+    """
+    \b
+    Create a new out-of-tree module
+
+    The argument MODULE-NAME overrides the current module's name (normally is autodetected).
+    """
+    args = DictToObject(kwargs)
+    try:
+        ModToolNewModule().run(args)
```

Comment:
just as a heads-up, the core-classes will need to take the actual params instead of a namespace object to make their use as API classes easier